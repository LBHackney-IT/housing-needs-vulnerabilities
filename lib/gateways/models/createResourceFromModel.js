import Resource from 'lib/domain/resource';

// .filter() will ignore null or undefined values
const filterEmptyValues = item => item;

export function createResourceFromModel({ id, fields }) {
  const websites = [fields['Website 1'], fields['Website 2']].filter(
    filterEmptyValues
  );

  return new Resource({
    id: id,
    name: fields.Name,
    description: fields.Description,
    websites,
    address: fields.Address,
    tags: fields['Vulnerability snapshot tags']
  });
}
