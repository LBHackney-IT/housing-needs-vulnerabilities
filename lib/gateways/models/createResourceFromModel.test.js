import { createResourceFromModel } from './createResourceFromModel';

describe('createResourceFromModel', () => {
  it('maps properties from an Airtable record', () => {
    const record = {
      id: 'dakjhksd8283',
      fields: {
        Name: 'Resource name',
        Description: 'This is a description',
        'Website 1': 'https://www.example.com',
        Address: '42 Wallaby Way, Sydney',
        Tags: ['one', 'two', 'three']
      }
    };

    expect(createResourceFromModel(record)).toMatchSnapshot();
  });
});
