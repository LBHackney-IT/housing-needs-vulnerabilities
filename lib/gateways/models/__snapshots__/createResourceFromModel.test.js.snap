// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createResourceFromModel maps properties from an Airtable record 1`] = `
Resource {
  "address": "42 Wallaby Way, Sydney",
  "description": "This is a description",
  "id": "dakjhksd8283",
  "name": "Resource name",
  "tags": Array [],
  "websites": Array [
    "https://www.example.com",
  ],
}
`;
